### DISCUSSION TITLE
Making shrubs less painful

### DISCUSSION THREAD

**Discussion by scarf005** [cataclysmbn/Cataclysm-BN#2179] [General]
Currently shrubs crush your vehicle for unknown reasons.

1. Make kenan's [shrub it off](https://github.com/Kenan2000/BrightNights-Structured-Kenan-Modpack/tree/master/Kenan-BrightNights-Structured-Modpack/High-Maintenance-Huge-Mods/shrub_it_off) the default
Pros: the simplest
Cons: What about any other flags with `Shrub`?, for example, peanut shrubs

2. Make terrain with `Shrub` flag crushable in vehicle
Pros: future proof
Cons: need a bit of CPP busywork. A bit of help for general vehicle collision algorithm would be useful.

**Comment by Proxiehunter** [comment]
I think you and I are associating the word shrub with a bush like plant you might be able to grow in a small pot on your porch while the game is defining these car wrecking shrubs more along [these lines](https://en.wikipedia.org/wiki/Shrub#/media/File:Cytisus_scoparius2.jpg). However even that is not likely to wreck _panels_ like I've seen in game shrubs do. You probably _could_ push through it with a car but you'd fuck your undercarriage, axles, tires and wheels. 

Last time I ran through a shrub without enabling shrub it off though those weren't the parts of the car damaged. The shrubs were breaking my headlights and front panels like I was ramming a full grown _tree_ or another car and that just isn't right.

**Comment by chaosvolt** [comment]
Additional options would be:
1. Adjusting how bash code gets vehicle damage from how hard it is to bash the terrain down. If it DOESN'T do that, i.e. if fragile terrain deals exactly as much damage to vehicles as super-hard walls, then fixing that would be a must-have.
2. Alternatively, if we do have bash-resist scaling for vehicle damage and it feels like the scaling isn't too wonky, then tweaking the bash resistance of shrubs is also an option.

**Reply by scarf005** [reply]
for part 2, dealing 50% less damage for terrain with `SHRUB` flag may also work

**Reply by chaosvolt** [reply]
I guess it depends on how does vehicle damage for ramming things work? Does it correctly scale the backlash with how hard the terrain is to bash, or is it solely controlled by flag hackery?
